network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -2008,6 +2038,94 @@ private <T extends IdentifiableAttributes> void deletePermanentLimits(UUID netwo
         resourceIdsByVariant.forEach((k, v) -> deletePermanentLimits(networkUuid, k, v));
     }
 
+    // Regulation Points
+    public void insertRegulationPoints(Map<OwnerInfo, RegulationPointAttributes> regulationPoints) {
+        try (var connection = dataSource.getConnection()) {
+            try (var preparedStmt = connection.prepareStatement(QueryCatalog.buildInsertRegulationPointsQuery())) {
+                List<Object> values = new ArrayList<>(6);
+                List<Map.Entry<OwnerInfo, RegulationPointAttributes>> list = new ArrayList<>(regulationPoints.entrySet());
+                for (List<Map.Entry<OwnerInfo, RegulationPointAttributes>> subUnit : Lists.partition(list, BATCH_SIZE)) {
+                    for (Map.Entry<OwnerInfo, RegulationPointAttributes> attributes : subUnit) {
+                        values.clear();
+                        values.add(attributes.getKey().getNetworkUuid());
+                        values.add(attributes.getKey().getVariantNum());
+                        values.add(attributes.getValue().getRegulatedEquipmentId());
+                        values.add(attributes.getValue().getRegulationMode());
+                        values.add(attributes.getValue().getLocalTerminal().getConnectableId());
+                        values.add(attributes.getValue().getLocalTerminal().getSide());
+                        values.add(attributes.getValue().getRegulatingTerminal().getConnectableId());
+                        values.add(attributes.getValue().getRegulatingTerminal().getSide());
+                        bindValues(preparedStmt, values, mapper);
+                        preparedStmt.addBatch();
+                    }
+                    preparedStmt.executeBatch();
+                }
+            }
+        } catch (SQLException e) {
+            throw new UncheckedSqlException(e);
+        }
+    }
+
+    public Map<OwnerInfo, RegulationPointAttributes> getRegulationPoints(UUID networkUuid, int variantNum, ResourceType type) {
+        try (var connection = dataSource.getConnection()) {
+            var preparedStmt = connection.prepareStatement(QueryCatalog.buildRegulationPointsQuery());
```

Comment:
Here I am wondering if I filter the regulation point associated with network element ids or I get all regulating points and I filter when I associate the regulating point with the network element

Reply:
For performance reason, at least, maybe it's better to filter with the network element kind (but, for that, you have to add the regulatedEquipmentType in regulationpoint table ...)
Otherwise we will always get all regulation points in the network, even when we want to get all equipments of a certain kind (see getStaticVarCompensators method)

Reply:
finaly I add to add the equipment type to the database because the primary key was not enough we can hava different elements with same id